File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/reference/part-one-php/src/Dashboard/Playground.php
<?php

namespace App\Dashboard;

use Grafana\Foundation\Common\ReduceDataOptionsBuilder;
use Grafana\Foundation\Dashboard as SDKDashboard;
use Grafana\Foundation\Dashboard\DashboardBuilder;
use Grafana\Foundation\Dashboard\DashboardCursorSync;
use Grafana\Foundation\Logs;
use Grafana\Foundation\Stat;
use Grafana\Foundation\Text;
use Grafana\Foundation\Timeseries;

class Playground
{
    public static function create(): SDKDashboard\Dashboard
    {
        $builder = (new DashboardBuilder(title: 'Test dashboard'))
            ->uid('test-dashboard')
            ->tags(['test', 'generated'])
            ->readonly()
            ->time('now-30m', 'now')
            ->tooltip(DashboardCursorSync::crosshair())
            ->refresh('10s')
        ;

        $builder
            ->withPanel(self::prometheusVersionStat())
            ->withPanel(self::descriptionText())
            ->withPanel(self::unfilteredLogs())
            ->withPanel(self::prometheusGoroutinesTimeseries())
        ;

        return $builder->build();
    }

    private static function prometheusVersionStat(): Stat\PanelBuilder
    {
        return Common::statPanel()
            ->title('Prometheus version')
            ->withTarget(
                Common::instantPrometheusQuery('prometheus_build_info{}')
            )
            ->transparent(true)
            ->datasource(Common::prometheusDatasourceRef())
            ->reduceOptions(
                (new ReduceDataOptionsBuilder())
                ->calcs(['last'])
                ->fields('/^version$/')
            )
        ;
    }

    private static function descriptionText(): Text\PanelBuilder
    {
        return Common::textPanel('Text panels are supported too! Even with *markdown* text :)')
            ->transparent(true)
        ;
    }

    private static function unfilteredLogs(): Logs\PanelBuilder
    {
        return Common::logPanel()
            ->title('Logs')
            ->datasource(Common::lokiDatasourceRef())
            ->withTarget(
                Common::lokiQuery('{job="app_logs"}')
            )
        ;
    }

    private static function prometheusGoroutinesTimeseries(): Timeseries\PanelBuilder
    {
        return Common::timeseriesPanel()
            ->title('Prometheus goroutines')
            ->withTarget(
                Common::prometheusQuery('go_goroutines{job="prometheus"}')
            )
            ->datasource(Common::prometheusDatasourceRef())
        ;
    }
}

File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/reference/part-one-java/src/main/java/lab/Playground.java
package lab;

import com.grafana.foundation.common.ReduceDataOptionsBuilder;
import com.grafana.foundation.dashboard.*;
import java.util.List;
import static lab.Common.*;

public class Playground {
    public static DashboardBuilder dashboard() {
        return new DashboardBuilder("Test dashboard").
                uid("test-dashboard").
                tags(List.of("generated", "test")).
                readonly().
                time(new DashboardDashboardTimeBuilder().
                        from("now-30m").
                        to("now")
                ).
                tooltip(DashboardCursorSync.CROSSHAIR).
                refresh("10s").
                withPanel(prometheusVersionStat()).
                withPanel(descriptionText()).
                withPanel(unfilteredLogs()).
                withPanel(prometheusGoroutinesTimeseries());
    }

    private static com.grafana.foundation.stat.PanelBuilder prometheusVersionStat() {
        return statPanel().
                title("Prometheus version").
                withTarget(
                        instantPrometheusQuery("prometheus_build_info{}")
                ).
                datasource(prometheusDatasourceRef()).
                transparent(true).
                reduceOptions(new ReduceDataOptionsBuilder().
                        calcs(List.of("last")).
                        fields("/^version$/")
                );
    }

    private static com.grafana.foundation.text.PanelBuilder descriptionText() {
        return textPanel("Text panels are supported too! Even with *markdown* text :)").
                transparent(true);
    }

    private static com.grafana.foundation.logs.PanelBuilder unfilteredLogs() {
        return logPanel().
                title("Logs").
                withTarget(lokiQuery("{job=\"app_logs\"}"));
    }

    private static com.grafana.foundation.timeseries.PanelBuilder prometheusGoroutinesTimeseries() {
        return timeseriesPanel().
                title("Prometheus goroutines").
                withTarget(prometheusQuery("go_goroutines{job=\"prometheus\"}")).
                datasource(prometheusDatasourceRef());
    }
}
